### PR TITLE
Vender.sol basics

### PR DESCRIPTION
**Pull request by jake-ritz11**

### CODE REVIEW

**Review by ahantke1 — DISMISSED** [review]
Looks like a good start to me! Only need to add the actual spin logic (however many functions that may be) now.

File: contracts/Vender.sol [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  /**
+  * @notice Allow the owner of the contract to withdraw ETH
+  */
+  function withdraw() public onlyOwner {
+    uint256 ownerBalance = address(this).balance;
+    require(ownerBalance > 0, "Owner has not balance to withdraw");
+
+    (bool sent,) = msg.sender.call{value: address(this).balance}("");
```
Comment (amir-raissi) [inline contracts/Vender.sol:53]: Not sure if this comma will throw error

**Comment by amir-raissi** [comment]
Looks good to me besides that one comment which may or may not be an issue. 

As for the amount of ETH to charge per token I think it should be in relation to the jackpot or we could totally just set it statically at like .00X eth. Something to think about for sure tho

**Review by amir-raissi — APPROVED** [review]
Think this is good base, need to add/think about the things i described above.

File: contracts/Vender.sol [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  event BuyTokens(address buyer, uint256 amountOfETH, uint256 amountOfTokens);
+
+  constructor(address tokenAddress) {
+    playToken = PlayToken(tokenAddress);
+  }
+
+  /**
+  * @notice Allow users to buy token for ETH
+  */
+  function buyTokens() public payable returns (uint256 tokenAmount) {
```
Comment (ahantke1) [inline contracts/Vender.sol:27]: Not working

Comment (ahantke1) [reply contracts/Vender.sol:27]: Testing on both hardhat and remix-- tokens are not being received by sender for some reason.

**Review by ahantke1 — CHANGES_REQUESTED** [review]

File: contracts/Vender.sol (hunk 1 above)
Comment (ahantke1) [reply contracts/Vender.sol:52]: it dont

**Comment by ahantke1** [comment]
Fails autotest and manual testing on remix: 

Issue with this line - 
32: uint256 vendorBalance = playToken.balanceOf(address(this));

**Review by ahantke1 — APPROVED** [review]

**Pull request merged**